File: SimplifyPath.py
----------------------------------------------------------------------------------------------------------------------------------------------------------------------

SIMPLIFY PATH

Question Link:
https://leetcode.com/problems/simplify-path/submissions/
        
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
path="/home/../"
stack=[]
        
for directory in path.split("/"):
    if directory=='':
        pass
    elif  directory==".":
        pass
    elif  directory=="..":
        if stack:
            stack.pop()
        else:
            pass
    else:
        stack.append(directory)
                
                
                
print( "/" + "/".join(stack)  )

File: SimplifyPath.java
class Solution {
    public String simplifyPath(String path) {
        
        Stack<String> st=new Stack<String>();
        String[] words=path.split("/");
        
        
        for(String w:words){
            if(!st.isEmpty() &&  w.equals(".."))
                st.pop();
            else if(!w.equals("..") && !w.equals(".") &&  !w.equals(""))
                st.push(w);
        }
            
        ArrayList<String> list=new ArrayList(st);
      
        return "/"+String.join("/",list);
        
      
}
    }